docs/design/proposals/encryption-with-vault-sa.md

```diff
@@ -0,0 +1,84 @@
+# Encrypted Volumes with a ServiceAccount per Tenant
+
+## Requirements
+
+- Hashicorp Vault with Kubernetes Auth
+- each Tenant (=Namespace) has a ServiceAccount to connect to Vault
+- each Tenant can override Vault configuration options (like `BackendPath`)
+
+## Similarities with available KMS implementations
+
+Ceph-CSI provides several KMS implementations with different functionalities.
+Two implementations use Hashicorp Vault and could be used as a base for a new
+KMS implementation. Or, if changes would be minimal, a configuration option to
+one of the implementations can be added.
+
+Different KMS implementations and their configurable options can be found at
+[`csi-kms-connection-details.yaml`](https://github.com/ceph/ceph-csi/blob/devel/examples/kms/vault/csi-kms-connection-details.yaml).
+
+### VaultTokensKMS
+
+- global configuration in the storage deployment namespace
+- optional configuration per Tenant, allows for overriding Vault options
+- fetches a Secret from the Tenants (volume owner) namespace
+
+An example of the per Tenant configuration options are in
+[`tenant-config.yaml`](https://github.com/ceph/ceph-csi/blob/devel/examples/kms/vault/tenant-config.yaml)
+and
+[`tenant-token.yaml`](https://github.com/ceph/ceph-csi/blob/devel/examples/kms/vault/tenant-token.yaml).
+
+Implementation is in [`vault_tokens.go`](https://github.com/ceph/ceph-csi/blob/devel/internal/util/vault_tokens.go).
+
+### Vault
+
+- uses Kubernetes Auth with single service account (aka storage admin account)
+
+Implementation is in [`vault.go`](https://github.com/ceph/ceph-csi/blob/devel/internal/util/vault.go).
+
+## Extension or New KMS implementation
+
+Normally ServiceAccounts are provided by Kubernetes in the containers
+filesystem. This only allowas a single ServiceAccount and is static for the
+lifetime of the Pod. Ceph-CSI runs in the namespace of the storage
+administrator, and has access to the single ServiceAccount linked in the
+deployments of the Pods.
+
+For Ceph-CSI to dynamically use ServiceAccounts from tenants, the following
+steps need to be taken:
+
+- use the owner of the volume to identify the tenant
```

Comment:
can you please elaborate what is meant by `owner` here?

Reply:
ok